File: java-to-clj-example/person.clj
(ns kaunas-jug.example.person
  (:require [clojure.data.csv :as csv]
            [clojure.java.io :as io]
            [clj-time.core :as t]
            [clj-time.format :as f]
            [clj-time.local :as l])
  (:import [jug.kaunas Person]))

(defn read-data [fpath]
  (let [reader (io/reader fpath)
        [header & body] (csv/read-csv reader)]
    body))

(defn parse-person
  [[fname lname date-of-birth hobbies]]
  (let [age (-> (f/parse (f/formatter "MM/dd/yyyy") date-of-birth)
                (t/interval (l/local-now))
                (t/in-years))]
    (new Person fname lname age (clojure.string/split hobbies #";"))))

(defn read-persons [fpath]
  (map parse-person
       (read-data fpath)))

File: java-to-clj-example/java/Person.java
package jug.kaunas;

import java.util.List;

public class Person {
    private final String firstName;
    private final String lastName;
    private short age;
    private List<String> hobbies;
    

    public Person(String firstName, String lastName, short age, List<String> hobbies){
        this.firstName = firstName;
        this.lastName = lastName;
        this.age = age;
        this.hobbies = hobbies;
    }

    public void greet(){
        System.out.printf("Hello, %s\n", getName());
    }

    public void printHobbies(){
        System.out.printf("%s hobbies: \n", getName());
        for(String hobby : hobbies){
            System.out.println(hobby);
        }
    }

    public String getName(){
        return firstName + " " + lastName;
    }

    @Override
    public String toString(){
        return String.format("<Person firstName: '%s' lastName: '%s' age: '%d'>", firstName, lastName, age);
    }
}
